### PR TITLE
Add Poetry support

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
following  https://github.com/jfrog/jfrog-cli/issues/1410 request

### CODE REVIEW

File: utils/coreutils/techutils.go [hunk 1]
```diff
@@ -67,6 +68,10 @@ var technologiesData = map[Technology]TechData{
 		PackageType: "pypi",
 		indicators:  []string{"pipfile", "pipfile.lock"},
 	},
+	Poetry: {
```
Comment (sverdlov93) [inline utils/coreutils/techutils.go:71]: Add exclude slice like Npm has under Pip for pipenv and poetry indicators

File: xray/audit/python/python_test.go [hunk 2]
```diff
@@ -58,3 +59,22 @@ func TestBuildPipenvDependencyList(t *testing.T) {
...
+	rootNodes, err := BuildDependencyTree(pythonutils.Poetry)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NotEmpty(t, rootNodes)
+
+	// Test child module
+	childNode := audit.GetAndAssertNode(t, rootNodes, "pytest:5.4.3")
+	// Test sub child module
+	transitiveChildNode := audit.GetAndAssertNode(t, childNode.Nodes, "packaging:21.3")
```
Comment (sverdlov93) [inline xray/audit/python/python_test.go:77]:
```suggestion
if assert.NotNil(t, childNode) {
	transitiveChildNode := audit.GetAndAssertNode(t, childNode.Nodes, "packaging:21.3")
	......
```
so it won't panic if childnodes is nil

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -22,6 +22,9 @@ jobs:
           
       - name: Install pipenv
         run: pip3 install pipenv
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
```
Comment (sverdlov93) [inline .github/workflows/test.yml:27]:
```suggestion
        uses: pip3 install poetry
```

File: utils/coreutils/techutils.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ const (
 	Go     = "go"
 	Pip    = "pip"
 	Pipenv = "pipenv"
+	Poetry = "poetry"
```
Comment (eyalbe4) [inline utils/coreutils/techutils.go:19]: @omerzi created a PR which unifies the package managers name across the entire code. Worth synching with him to make the merge easier.

File: xray/audit/python/python.go [hunk 5]
```diff
@@ -143,6 +144,19 @@ func runPythonInstall(tempDirPath string, pythonTool pythonutils.PythonTool) (re
 		if err != nil {
 			err = errorutils.CheckErrorf("pipenv install command failed: %s - %s", err.Error(), output)
 		}
+
```
Comment (eyalbe4) [inline xray/audit/python/python.go:147]: Redundant new-line.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**